resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -97,4 +97,17 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
             }
         });
     }
+
+    /**
+     * @param anyInt any integer value to covert
+     * @param len max limit of range
+     * @return integer in (0, len] range, using formula {@code (anyInt % len + anyInt) % len}
+     */
+    protected static int toIndex(int anyInt, int len) {
+        int index = anyInt % len;
+        if (index < 0) {
+            index = index + len;
```

Comment:
i % 5 + 5 = (i % 5) + 5 as % takes precedence, if that wasn't true then my output would be equivalent to i % 10 which would show on my example and it doesn't.

```
for (int i = -10; i < 0; i++) {
      final int index = i % 5 + 5;
      System.out.println("Without -1 for i = " + i + ", index = " + index);
      System.out.println("With -1 for i = " + i + ", index = " + (index - 1));
      System.out.println("With index function for = " + i + ", index = " + toIndex(i, 5));
    }
```
Result:
```
Without -1 for i = -10, index = 5
With -1 for i = -10, index = 4
With index function for = -10, index = 0
Without -1 for i = -9, index = 1
With -1 for i = -9, index = 0
With index function for = -9, index = 1
Without -1 for i = -8, index = 2
With -1 for i = -8, index = 1
With index function for = -8, index = 2
Without -1 for i = -7, index = 3
With -1 for i = -7, index = 2
With index function for = -7, index = 3
Without -1 for i = -6, index = 4
With -1 for i = -6, index = 3
With index function for = -6, index = 4
Without -1 for i = -5, index = 5
With -1 for i = -5, index = 4
With index function for = -5, index = 0
Without -1 for i = -4, index = 1
With -1 for i = -4, index = 0
With index function for = -4, index = 1
Without -1 for i = -3, index = 2
With -1 for i = -3, index = 1
With index function for = -3, index = 2
Without -1 for i = -2, index = 3
With -1 for i = -2, index = 2
With index function for = -2, index = 3
Without -1 for i = -1, index = 4
With -1 for i = -1, index = 3
With index function for = -1, index = 4
```